### PR TITLE
Remove changelog requirement from PR template

### PR DESCRIPTION
**Pull request by mrocklin**
See https://github.com/dask/dask/issues/3447

This should reduce friction for submitting PRs

- [ ] Tests added / passed
- [ ] Passes `flake8 dask`
- [ ] Fully documented, including `docs/source/changelog.rst` for all changes
      and one of the `docs/source/*-api.rst` files for new API

### CODE REVIEW

**Comment by mrocklin** [comment]
Merging in 24 hours if there are no further comments

**Comment by mrocklin** [comment]
I'll also bring this up at the [meeting today](http://dask.pydata.org/en/latest/support.html)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,4 +1,2 @@
 - [ ] Tests added / passed
 - [ ] Passes `flake8 dask`
-- [ ] Fully documented, including `docs/source/changelog.rst` for all changes
-      and one of the `docs/source/*-api.rst` files for new API
```
Comment (jcrist) [inline .github/PULL_REQUEST_TEMPLATE.md]: Perhaps leave the bit about documenting new api?

**Comment by mrocklin** [comment]
Yeah, I thought about that.  My experience has been that most PRs don't
involve documentation changes, and those that do have been large enough
that we're likely to mention it anyway.  I'm inclined to skip this in order
to reduce cognitive load on new contributors (which is really who these
checklists are for)

**Comment by jcrist** [comment]
Fine by me.

**Comment by jrbourbeau** [comment]
> I'm inclined to skip this in order to reduce cognitive load on new contributors (which is really who these checklists are for)

That sounds reasonable to me 👍

**Pull request merged**